learning_paths/api/v1/views.py

```diff
@@ -105,3 +118,233 @@ def get(self, request, learning_path_uuid):
+        ):
+            raise PermissionDenied
+
+        enrollment, created = LearningPathEnrollment.objects.get_or_create(
+            learning_path=learning_path,
+            user=user,
+            defaults={"is_active": True, "enrolled_at": datetime.now(timezone.utc)},
+        )
+        if created:
+            return Response(
+                LearningPathEnrollmentSerializer(enrollment).data,
+                status=status.HTTP_201_CREATED,
+            )
+        if enrollment.is_active:
+            return Response(
+                {"detail": "Enrollment exists."}, status=status.HTTP_409_CONFLICT
+            )
+
+        enrollment.is_active = True
+        enrollment.enrolled_at = datetime.now(timezone.utc)
+        enrollment.save()
+        return Response(LearningPathEnrollmentSerializer(enrollment).data)
+
+    def delete(self, request, learning_path_id):
+        """
+        Unenroll a learner from a learning path.
+
+        Staff/admin can unenroll anyone with the username query param.
+        Learners can self-unenroll if settings.LEARNING_PATHS_ALLOW_SELF_UNENROLLMENT is True.
+        """
+        learning_path = get_object_or_404(LearningPath, uuid=learning_path_id)
+        username = request.GET.get("username")
+
+        enrollment = LearningPathEnrollment.objects.filter(
+            learning_path=learning_path, is_active=True
+        )
+        if username:
+            enrollment = enrollment.filter(user__username=username).first()
+        else:
+            enrollment = enrollment.filter(user=request.user).first()
+
+        if not enrollment:
+            return Response(
+                {"detail": "No enrollment."}, status=status.HTTP_404_NOT_FOUND
+            )
+
+        if (
+            not request.user.is_staff
+            and not request.user.is_superuser
+            and request.user != enrollment.user
+        ):
+            raise PermissionDenied
+
+        if (
+            not settings.LEARNING_PATHS_ALLOW_SELF_UNENROLLMENT
+            and request.user == enrollment.user
+        ):
+            raise PermissionDenied
+
+        enrollment.is_active = False
+        enrollment.save()
+        return Response(
+            LearningPathEnrollmentSerializer(enrollment).data,
+            status=status.HTTP_204_NO_CONTENT,
+        )
+
+
+class FetchEnrollmentsView(APIView):
+    """
+    A view to fetch learning path enrollments for the authenticated user.
+    """
+
+    permission_classes = [IsAuthenticated]
+
+    def get(self, request):
+        """
+        Handle GET requests to fetch enrollments.
+        """
+        if request.user.is_staff or request.user.is_superuser:
+            enrollments = LearningPathEnrollment.objects.all()
+        else:
+            enrollments = LearningPathEnrollment.objects.filter(user=request.user)
+
+        return Response(LearningPathEnrollmentSerializer(enrollments, many=True).data)
+
+
+class BulkEnrollAPIView(APIView):
```

Comment:
```suggestion
class BulkEnrollView(APIView):
```

Nit: it didn't match the other names.